bazel_codegen/lib/bazel_codegen.dart

```diff
@@ -0,0 +1,82 @@
+import 'dart:async';
+
+import 'package:build/build.dart';
+
+import 'src/arg_parser.dart';
+import 'src/run_phases.dart';
+
+typedef Builder BuilderFactory(List<String> args);
+
+/// Wrap [builder] as a [BuiderFactory] that expects no arguments.
+///
+/// Soon this will throw if any unexpected arguments are passed.
+BuilderFactory noArgs(Builder builder) => (List<String> args) {
+      if (args?.isNotEmpty ?? false) {
```

Comment:
This is a bit hard to reason about, what about `if (args?.isNotEmpty == true) {`?